src/state/transactions/updater.tsx

```diff
@@ -24,9 +26,22 @@ export function toSerializableReceipt(receipt: TransactionReceipt): Serializable
   }
 }
 
+/**
+ * Returns the time elapsed between page load and now,
+ * if the time-to-swap mark doesn't already exist.
+ *
+ * We only log the time-to-swap metric for the first swap of a session.
+ */
+function getElapsedTime(): number {
+  const timeToSwap = performance.mark('time-to-swap')
+  return timeToSwap.startTime
+}
+
 export default function Updater() {
+  const analyticsContext = useTrace()
   const { chainId } = useWeb3React()
   const addPopup = useAddPopup()
+  const [timeToSwap, setTimeToSwap] = useState<number | undefined>(undefined)
```

Comment:
This doesn't need to be a part of the React lifecycle. In other words, you can move this to a file-level variable, `let hasReportedTimeToSwap = false`.